spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/VectorizedSparkParquetReaders.java

```diff
@@ -56,7 +58,8 @@ public static ColumnarBatchReader buildReader(
       Schema expectedSchema,
       MessageType fileSchema,
       Map<Integer, ?> idToConstant,
-      DeleteFilter<InternalRow> deleteFilter) {
+      DeleteFilter<InternalRow> deleteFilter,
+      BufferAllocator bufferAllocator) {
     return (ColumnarBatchReader)
         TypeWithSchemaVisitor.visit(
             expectedSchema.asStruct(),
```

Comment:
Another Change to allow passing through the allocator

Reply:
should we make this new factory method package private and add the `@VisibleFoTesting` annotation?